nodejs/extensions.bzl

```
@@ -1,12 +1,19 @@
 "extensions for bzlmod"
 
-load(":repositories.bzl", "DEFAULT_NODE_VERSION", "nodejs_register_toolchains")
+load(":repositories.bzl", "DEFAULT_NODE_REPOSITORY", "DEFAULT_NODE_VERSION", "nodejs_register_toolchains")
 
 def _toolchain_extension(module_ctx):
     registrations = {}
     for mod in module_ctx.modules:
         for toolchain in mod.tags.toolchain:
+            if toolchain.name != DEFAULT_NODE_REPOSITORY and not mod.is_root:
+                fail("Only the root module may provide a name for the node toolchain.")
+
             if toolchain.name in registrations.keys():
+                if toolchain.name == DEFAULT_NODE_REPOSITORY:
+                    # Prioritize the root-most registration of the default node toolchain version and
+                    # ignore any further registrations (modules are processed depth-first)
```

```suggestion
                    # Prioritize the root-most registration of the default node toolchain version and
                    # ignore any further registrations (modules are processed starting from the root module)
```
The "depth-first" part doesn't seem right as the [docs specify breadth-first iteration](https://docs.bazel.build/versions/5.0.0/skylark/lib/module_ctx.html#modules):
> The iteration order of this dictionary is guaranteed to be the same as breadth-first search starting from the root module.

---

Oops, typo. Thanks for catching.